include/OscillatorConstants.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * OscillatorConstants.h - declaration of constants used in  Oscillator and SampleBuffer
+ * for band limited wave tables
+ *
+ * Copyright (c) 2018      Dave French	<dave/dot/french3/at/googlemail/dot/com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef OSCILLATORCONSTANTS_H
+#define OSCILLATORCONSTANTS_H
+
+#include <array>
+
+class OscillatorConstants
+{
+public:
+	// Minimum size of table to have all audible bands for midi note 1 (i.e. 20 000 Hz / 8.176 Hz)
+	static const int WAVETABLE_LENGTH = 2446;
```

Comment:
Can you please rewrite this as `static const int WAVETABLE_LENGTH = floorf(MAX_FREQ / 8.176)`? That way, we are safe if anyone changes `MAX_FREQ` in the future and does not look at this variable.

Reply:
Or maybe even something like this:
```cpp
constexpr int WAVETABLE_LENGTH = static_cast<int>(MAX_FREQ / 8.176);
```

Reply:
Huh, I did not even know that something like `constexpr` existed. I still learn something new every day. :)) I will use it as suggested.